### DISCUSSION TITLE
Passing all arguments to another external command ("$@")

### DISCUSSION THREAD

**Discussion by robaca** [nushell/nushell#7457] [Q&A] [answered]
Hi, 

how would I pass parts or all parameters of a script to another external command, independent of their type (also flags):

I try to achieve something like this: 
```
#!/usr/bin/env bash
SOME_VAR=$1
shift 1
other-command "$@"
```

Creating a custom command with rest parameter is failing for me for flags.

**Comment by rgwood** [comment]
I'm not sure if we currently have a good way to do flags. I think this works for regular arguments but *not* flags:
```
def main [...args] {
  other-command $args
}
```

We may need to add support for something like `--` to Nu.

Please raise an issue if you have time.

**Reply by yhm-amber** [reply]
Coincidentally, I've raise an `question` issue about `--` at #7397 😃

(But I dont know why can't I search it by keyword `--` ...)

---

btw, if nushell support `"$@"`, the sub command may can be define like this:

~~~ nu
def foo [...]
{
    def bar [...] { ... }
    
    "$@"
}

foo bar ...
~~~

then `foo` like a package :P .

**Comment by ErichDonGubler** [comment]
Related, but not the same: https://github.com/nushell/nushell/issues/10509

**Comment by samoylovfp** [comment, marked as answer]
There is a flag [`--wrapped`](https://www.nushell.sh/commands/docs/export_def.html) that makes the `...rest` parameter capture flags as well

**Reply by KawaiiSelbst** [reply]
It's good to have it. But it doesn't help in situations where you want to do something like 

```shell
nu -c `
  let result = ((zoxide query --exclude $env.PWD ...$rest) | str replace "/", "//") 
  lf -remote $"send 1492 cd '($result)'"
  lf -remote $"send 1492 echo '($result)'"
` [$args]
```

some programs are able to execute user scripts only in this format, of course maybe it is more a question of such programs, although as I understand it is quite a classical approach, because in unix bash is used by default and all arguments are always available by default using `$@`.
Because there is no way to accept all arguments in nu, it becomes simply impossible to use nu with them
